### PR TITLE
Validation testing for chm_diags (from mo_chm_diags)

### PR DESCRIPTION
**Pull request by jaelynlitz**
work in progress, most of the typing has been done, but haven't had the chance to fine tune/fix all compile errors.

cc @mjs271 @odiazib

### CODE REVIEW

**Comment by jeff-cohere** [comment]
Hey @mjs271 and @odiazib . I might be able to take this one to the finish line if you're both occupied. Let me know.

**Comment by odiazib** [comment]
> Hey @mjs271 and @odiazib . I might be able to take this one to the finish line if you're both occupied. Let me know.

Hi Jeff, please do.

**Comment by jeff-cohere** [comment]
Now the test is running on CPU and GPU, but it's not passing. I suspect there's something wrong that's easy to understand. Also, I think we might need to augment the dataset for `mo_chm_diags` to add the `area` parameter as an input, since it's an "inout" parameter in the subroutine of interest.

File: src/mam4xx/mo_chm_diags.hpp [hunk 1]
```diff
@@ -64,6 +72,283 @@ void het_diags(
...
+    if (name[i] != pattern[i])
+      return false;
+  }
+  return true;
+}
+
+} // namespace
+
+//========================================================================
+// TODO: toth and tcly vars not actually used in the function...
```
Comment (odiazib) [inline src/mam4xx/mo_chm_diags.hpp:103]: Is this comment still valid?

**Review by odiazib — APPROVED** [review]
Looks great!

**Comment by jaelynlitz** [comment]
Thanks for finishing this off @jeff-cohere ! 🙌

**Pull request merged**